tools/export/cdt/__init__.py

```diff
@@ -29,11 +30,19 @@ def generate(self):
         if not exists(join(self.export_dir,'eclipse-extras')):
             makedirs(join(self.export_dir,'eclipse-extras'))
 
+        templates = ['%s.tmpl' % (self.target.lower())] + \
+            ['%s.tmpl' % (label.lower()) for label
+            in self.toolchain.target.extra_labels] + \
+            ['%s.tmpl' % 'pyocd_settings']
+
+        for templatefile in templates:
+            try:
+                self.gen_file('cdt/%s' % templatefile, ctx, join('eclipse-extras',
+                           '{target}_{project}_{launch}.launch'.format(target=self.target,
+                                                                    project=self.project_name, launch=templatefile)))
+            except TemplateNotFound:
```

Comment:
Just thought of this: without target-specific templates, there is no chance of missing templates. Could you get rid of the try-except here too?